velox/expression/tests/SparkExpressionRunnerTest.cpp

```diff
@@ -0,0 +1,195 @@
+DEFINE_string(
+    store_result_path,
+    "",
+    "Directory path for storing the results of evaluating SQL expression or "
+    "query in common, simplified or query modes.");
+
+DEFINE_string(
+    fuzzer_repro_path,
+    "",
+    "Directory path where all input files generated by ExpressionVerifier are "
+    "expected to reside. For more details on which files and their names are "
+    "expected, please checkout the ExpressionVerifier class. Any file paths "
+    "already specified via a startup flag will take precedence.");
+
+DEFINE_bool(
+    find_minimal_subexpression,
+    false,
+    "Automatically seeks minimum failed subexpression on result mismatch");
+
+static std::string checkAndReturnFilePath(
+    const std::string_view& fileName,
+    const std::string& flagName) {
+  auto path = fmt::format("{}/{}", FLAGS_fuzzer_repro_path, fileName);
+  if (fs::exists(path)) {
+    LOG(INFO) << "Using " << flagName << " = " << path;
+    return path;
+  } else {
+    LOG(INFO) << "File for " << flagName << " not found.";
+  }
+  return "";
+}
+
+static void checkDirForExpectedFiles() {
+  LOG(INFO) << "Searching input directory for expected files at "
+            << FLAGS_fuzzer_repro_path;
+
+  FLAGS_input_path = FLAGS_input_path.empty()
+      ? checkAndReturnFilePath(
+            test::ExpressionVerifier::kInputVectorFileName, "input_path")
+      : FLAGS_input_path;
+  FLAGS_result_path = FLAGS_result_path.empty()
+      ? checkAndReturnFilePath(
+            test::ExpressionVerifier::kResultVectorFileName, "result_path")
+      : FLAGS_result_path;
+  FLAGS_sql_path = FLAGS_sql_path.empty()
+      ? checkAndReturnFilePath(
+            test::ExpressionVerifier::kExpressionSqlFileName, "sql_path")
+      : FLAGS_sql_path;
+  FLAGS_lazy_column_list_path = FLAGS_lazy_column_list_path.empty()
+      ? checkAndReturnFilePath(
+            test::ExpressionVerifier::kIndicesOfLazyColumnsFileName,
+            "lazy_column_list_path")
+      : FLAGS_lazy_column_list_path;
+  FLAGS_complex_constant_path = FLAGS_complex_constant_path.empty()
+      ? checkAndReturnFilePath(
+            test::ExpressionVerifier::kComplexConstantsFileName,
+            "complex_constant_path")
+      : FLAGS_complex_constant_path;
+}
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  // Calls common init functions in the necessary order, initializing
+  // singletons, installing proper signal handlers for better debugging
+  // experience, and initialize glog and gflags.
+  folly::Init init(&argc, &argv);
+
+  if (!FLAGS_fuzzer_repro_path.empty()) {
+    checkDirForExpectedFiles();
+  }
+
+  if (FLAGS_sql.empty() && FLAGS_sql_path.empty()) {
+    std::cout << "One of --sql or --sql_path flags must be set." << std::endl;
+    exit(1);
+  }
+
+  auto sql = FLAGS_sql;
+  if (sql.empty()) {
+    sql = restoreStringFromFile(FLAGS_sql_path.c_str());
+    VELOX_CHECK(!sql.empty());
+  }
+
+  functions::sparksql::registerFunctions("");
```

Comment:
The only major difference with Presto expression fuzzer is the fact that the registered functions are different, will it make more sense to actually add a flag there to choose spark or presto registry rather than create this executable ?

Reply:
@kgpai 

I have these changes staged, please let me know what you think.

**Flag registration**

```cpp
DEFINE_string(
    engine,
    "presto",
    "Engine to use for SQL evaluation. Valid values are 'presto' and 'spark'."
    "Default value is 'presto'.");

```
**Flag parsing**

```cpp
  std::cout << FLAGS_engine << std::endl;
  if (FLAGS_engine == "spark") {
    functions::sparksql::registerFunctions("");
  } else if (FLAGS_engine == "presto") {
    functions::prestosql::registerAllScalarFunctions();
    aggregate::prestosql::registerAllAggregateFunctions();
  } else {
    std::cout << "Invalid value for --engine: " << FLAGS_engine << ". ";
    std::cout << "Valid values are: 'presto' and 'spark'." << std::endl;
    exit(1);
  }


```

**Output after local testing**

<img width="534" alt="image" src="https://github.com/facebookincubator/velox/assets/14911410/dd1a975c-5924-49b3-8f03-a0899847721b">

If this sounds good, I will incorporate these changes into ExpressionRunnerTest.cpp and remove the print statement, spark_expression_runner_test executable, and new libraries this commit introduced.

Reply:
@kgpai 

My only concern is potential linking errors in the future. When I started trying to implement jsons for SparkSQL I ran into some linking errors. I think that may have just been because I was new to velox's build structure and C++ at the time. I shelved that work for a while, and when I came back to it after a little more experience I was able to avoid those errors. 

I am still not super familiar with the best practices in regards to avoiding naming collisions, etc, in this codebase, so maybe linking both function libraries together isn't an issue at all. Could use some insight from someone with more experience here.